sections/first-things-first.md

```diff
@@ -128,40 +128,55 @@ Respond with standard HTTP error messages, with 4XX codes to indicate client (ca
 HTTP/1.1 400 Bad Request { "message" : "The request body had invalid json" }
 ```
 
-## Throttling & Limits
+## Throttling & Rate-Limiting
 
-The API currently throttles incoming requests to prevent abuse and ensure fair use. When throttling has been triggered, the affected callers will receive a HTTP response with status `429 Too Many Requests`.
+The API throttles incoming requests to prevent abuse and ensure fair use.
 
-### What to do when being throttled?
+When an application exceeds the rate limit, the API will return a standard HTTP `429 Too Many Requests` response.
 
-Our current recommendation is to implement a progressive back-off approach until your requests start processing normally. The psuedo-code below illustrate a simple scenario which you may adopt to suit your particular case.
+### How to handle being throttled
+
+The specific details of rate limit policies in effect may vary without notice. Application developers should implement detection and handling of rate-limiting behavior using `X-Ratelimit-*` headers described below.
+
+When a request from your application receives a `429 Too many Requests` response, you can check the returned HTTP headers to see your current rate limit status:
```

Comment:
@shyam-habarakada This reads as if the rate limit headers will only show up on the 429 response. I feel we should remove this sentence.